pkg/odo/cli/component/create.go

```diff
@@ -555,6 +647,10 @@ func NewCmdCreate(name, fullName string) *cobra.Command {
 	componentCreateCmd.Flags().StringSliceVarP(&co.componentPorts, "port", "p", []string{}, "Ports to be used when the component is created (ex. 8080,8100/tcp,9100/udp)")
 	componentCreateCmd.Flags().StringSliceVar(&co.componentEnvVars, "env", []string{}, "Environmental variables for the component. For example --env VariableName=Value")
 
+	if experimental.IsExperimentalModeEnabled() {
```

Comment:
Wondering if we should gate the `projectCmd.AddProjectFlag(componentCreateCmd)` line farther down for when the experimental mode is disabled. We won't need both `--project` and `--namespace` when working with devfile components

Reply:
I think the bottom line is we don't want to break the existing `s2i` support, so I am worry about gating `projectCmd.AddProjectFlag(componentCreateCmd)` may cause that kind of issue. I agree it's kind of redundant but it's not too bad to let them live together currently to avoid breaking changes. Also for `--namespace` flag we already had the specific description to make it only applies to `devfile` since it's generic kube (I think there is no `Project` definition in generic kube):
```
-n, --namespace string    Create devfile component under specific namespace
```

Reply:
According to the latest design, we share `--project` flag for s2i and devfile